src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,56 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static int ARRAY_SIZE = 10;
+    private K[] keyList;
+    private V[] valueList;
+    private int size;
+
+    public StorageImpl() {
+        keyList = (K[]) new Object[ARRAY_SIZE];
+        valueList = (V[]) new Object[ARRAY_SIZE];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex != -1) {
+            valueList[keyIndex] = value;
+        } else {
+            keyList[size] = key;
+            valueList[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex != -1) {
+            return valueList[getKeyIndex(key)];
```

Comment:
```suggestion
            return valueList[keyIndex];
```
Don’t call the method twice